### PR TITLE
Improvement for Cache mediator to honor the Cache-control headers and support the Age header

### PR DESCRIPTION
**Pull request by keerthu**
## Purpose
Improvement for Cache mediator to honor the Cache-control headers and support the Age header

## Goals
This improvement includes the following use cases:
- Honor no-store header.
- Honor no-cache header.
- Honor max-age header.
- Include Age header when returning the cached response.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/org/wso2/carbon/mediator/cache/ui/i18n/Resources.properties [hunk 1]
```diff
@@ -14,4 +14,6 @@ mediator.cache.protocoldetails=Protocol Details
 mediator.cache.protocol=Protocol Type
 mediator.cache.httpmethods=HTTP Methods To Cache (comma separated list)
 mediator.cache.headersToExclude=Headers To Exclude In Hash (comma separated list)
-mediator.cache.responsecodes=Response Codes (regex expression)
\ No newline at end of file
+mediator.cache.responsecodes=Response Codes (regex expression)
+mediator.cache.cacheControlEnabled=Enable Cache Control Headers
+mediator.cache.includeAgeHeader=Include Age Header
```
Comment (menakaj) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/org/wso2/carbon/mediator/cache/ui/i18n/Resources.properties]: Shall we add a new line here?

Comment (muthulee) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/org/wso2/carbon/mediator/cache/ui/i18n/Resources.properties]: New line.

File: components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CachableResponse.java [hunk 2]
```diff
@@ -322,4 +337,58 @@ public String getHttpMethod() {
...
+
+    /**
+     * This method sets whether an Age header needs to be included or not.
+     *
+     * @param addAgeHeaderEnabled Whether an Age header needs to be included or not.
+     */
+    public void setAddAgeHeaderEnabled(boolean addAgeHeaderEnabled) {
+        this.addAgeHeaderEnabled = addAgeHeaderEnabled;
+    }
+}
```
Comment (menakaj) [inline components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CachableResponse.java]: Add a new line here.

**Comment by menakaj** [comment]
It's better if we can add a better git commit message. As there are several format change commits, shall we squash them into one commit?
[](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)

**Comment by keerthu** [comment]
Closing this due to the conflicts.

**Pull request closed without merging**